src/app/qgsattributetabledialog.cpp

```diff
@@ -890,14 +890,18 @@ void QgsAttributeTableDialog::mActionAddAttribute_triggered()
     if ( mLayer->addAttribute( dialog.field() ) )
     {
       mLayer->endEditCommand();
+
+      if ( mLayer->displayExpression() == QStringLiteral( "'[Please define preview text]'" ) )
```

Comment:
Is this working?
Normally the dual view shouldn't be setting the layer display expression from there directly.
Otherwise the code should be adapted in dual view too.

At the same time, it would be worth making the text translated.

Reply:
As far as I can see, this text is set as a display expression in the dual view https://github.com/qgis/QGIS/blob/master/src/gui/attributetable/qgsdualview.cpp#L222 and it is not translated.

Reply:
yes as display expression for the dual view, but I am not sure it's set back on the layer
It depends on what `QgsFeatureListModel::setDisplayExpression` returns. This needs to be tested.

Reply:
thanks to the changes in https://github.com/qgis/QGIS/pull/47016 you can now test if it is empty
```suggestion
      if ( mLayer->displayExpression().isEmpty() )
```

Reply:
Thanks, changed.